velox/functions/sparksql/aggregates/MinMaxAggregate.cpp

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/aggregates/MinMaxAggregateBase.h"
+
+using namespace facebook::velox::functions::aggregate;
+
+namespace facebook::velox::functions::aggregate::sparksql {
+
+namespace {
+class TimestampMaxAggregate : public MaxAggregate<Timestamp> {
+ public:
+  explicit TimestampMaxAggregate(TypePtr resultType)
+      : MaxAggregate<Timestamp>(resultType) {}
+
+  void extractValues(char** groups, int32_t numGroups, VectorPtr* result)
+      override {
+    using BaseAggregate =
+        SimpleNumericAggregate<Timestamp, Timestamp, Timestamp>;
+    BaseAggregate::template doExtractValues<Timestamp>(
+        groups, numGroups, result, [&](char* group) {
+          auto ts = *BaseAggregate::Aggregate::template value<Timestamp>(group);
+          return Timestamp::fromMicros(ts.toMicros());
```

Comment:
Do we need to document in https://github.com/facebookincubator/velox/blob/main/velox/docs/functions/spark/aggregate.rst?

Reply:
Added, thanks.